backend/amd64/emit.mlp

```diff
@@ -418,47 +418,72 @@ let emit_local_realloc lr =
   emit_call (Cmm.global_symbol "caml_call_local_realloc");
   I.jmp (label lr.lr_return_lbl)
 
-(* Record calls to caml_ml_array_bound_error.
-   In -g mode we maintain one call to
-   caml_ml_array_bound_error per bound check site.  Without -g, we can share
+(* Record calls to caml_ml_array_bound_error and caml_ml_array_align_error.
+   In -g mode we maintain one call per bound check site.  Without -g, we can share
    a single call. *)
 
-type bound_error_call =
-  { bd_lbl: label;                      (* Entry label *)
-    bd_frame: label;                    (* Label of frame descriptor *)
-    bd_dbg: Debuginfo.t;
-    (* As for [gc_call]. *)
+type safety_check = Bound_check | Align_check
+
+type safety_check_failure = {
+    sc_lbl: label;              (* Entry label *)
+    sc_frame: label;            (* Label of frame descriptor *)
+    sc_dbg: Debuginfo.t;        (* As for [gc_call]. *)
+  }
+
+type safety_check_sites = {
+    mutable sc_sites: safety_check_failure list;
+    mutable sc_call: label;
   }
 
-let bound_error_sites = ref ([] : bound_error_call list)
-let bound_error_call = ref 0
+let bound_checks = { sc_sites = []; sc_call = 0 }
+let align_checks = { sc_sites = []; sc_call = 0 }
 
-let bound_error_label fdo dbg =
+let safety_error_label kind fdo dbg =
   if !Clflags.debug then begin
     let lbl_bound_error = new_label() in
```

Comment:
(That name feels a bit off after the refactoring.)

Reply:
Is `safety_check_failure_label` better?